pyartcd/pyartcd/pipelines/check_bugs.py

```diff
@@ -0,0 +1,230 @@
+        self.logger = runtime.logger
+        self.applicable_versions = []
+        self.blockers = {}
+        self.regressions = {}
+        self.slack_client = self.initialize_slack_client(runtime, channel)
+
+    @staticmethod
+    def initialize_slack_client(runtime: Runtime, channel: str):
+        if not channel.startswith('#'):
+            raise ValueError('Invalid Slack channel name provided')
+
+        slack_client = runtime.new_slack_client()
+        slack_client.bind_channel(channel)
+        return slack_client
+
+    async def run(self):
+        # Check applicable OCP versions
+        await self._check_applicable_versions()
+
+        # Find blocker bugs
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = []
+            for v in self.applicable_versions:
+                futures.append(executor.submit(self._find_blockers, v))
+            for f in futures:
+                try:
+                    self.blockers.update(f.result())
+                except TypeError:
+                    # In case no blockers have been found
+                    pass
+
+        # Find regressions
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = []
+            for v in self.applicable_versions:
+                futures.append(executor.submit(self._find_regressions, v))
+            for f in futures:
+                try:
+                    self.regressions.update(f.result())
+                except TypeError:
+                    # In case no regressions have been found
+                    pass
+
+        # Notify Slack
+        await self._slack_report()
+        self.logger.info('All done!')
+
+    async def _check_applicable_versions(self):
+        ga_info = {}
+        async with aiohttp.ClientSession() as session:
+            tasks = []
+            for v in self.versions:
+                tasks.append(asyncio.ensure_future(is_ga(v, session)))
+            responses = await asyncio.gather(*tasks)
+            ga_info = dict(zip(self.versions, responses))
+
+        self.applicable_versions = [v for v in self.versions if ga_info.get(v, True)]
+
+        if self.applicable_versions:
+            self.logger.info(f'Found applicable versions: {" ".join(self.applicable_versions)}')
+        else:
+            self.logger.warning('No applicable versions found')
+
+    def _find_blockers(self, version: str):
+        self.logger.info(f'Checking blocker bugs for Openshift {version}')
+
+        cmd = [
+            ELLIOTT_BIN,
+            f'--group=openshift-{version}',
+            f'--working-dir={version}-working',
+            'find-bugs',
+            '--mode=blocker',
+            '--report',
+            '--output=slack'
+        ]
+        self.logger.info(f'Executing command: {" ".join(cmd)}')
+
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        out, _ = process.communicate()
+        errcode = process.returncode
```

Comment:
mark the job as failed if one or more things fail. then we know we have to look.
optional, a retry of a failed command to reduce flakes. but we can get to that if there would be too many flakes

Reply:
Done here: https://github.com/openshift/aos-cd-jobs/pull/3003/commits/281e546c8bd9475dcb2d19aa2d7f8cfbe46d0f1c
I'd leave retries for later, if we see we actually need this